scene/2d/path_2d.cpp

```diff
@@ -319,8 +319,24 @@ void PathFollow2D::_bind_methods() {
 void PathFollow2D::set_offset(float p_offset) {
 
 	offset = p_offset;
-	if (path)
+	if (path) {
+		if (path->get_curve().is_valid() && path->get_curve()->get_baked_length()) {
+			float path_length = path->get_curve()->get_baked_length();
+
+			if (loop) {
+				while (offset > path_length)
+					offset -= path_length;
+
+				while (offset < 0)
+					offset += path_length;
+
+			} else {
+				offset = CLAMP(offset, 0, path_length);
```

Comment:
Maybe `wrapf` could help solve #35485? @KoBeWi

Reply:
No, this is correct here, because we clamp the value if looping is disabled (which is irrelevant to #35485, because it uses loops).